code/datums/effects/neurotoxin.dm

```diff
@@ -21,9 +21,8 @@
 	/// Stamina damage per tick. Major balance number.
 	var/stam_dam = 7
 
-/datum/effects/neurotoxin/New(atom/thing)
-	..(thing)
-	cause_data = create_cause_data("neurotoxic gas")
+/datum/effects/neurotoxin/New(atom/thing, mob/from = null)
+	..(thing, from, effect_name)
```

Comment:
not entirely sure why this is preferable to

```suggestion
	. = ..()
	
	cause_data = create_cause_data(effect_name)
```

generally messing with the ..() signature can result in some particularly weird issues - but if this is fixing an issue i'm overlooking lmk

Reply:
The parent constructor takes four arguments, this constructor is taking two: but in the case of this we want the third argument to always be effect_name and we don't care about the fourth argument.

Also the cause_data should contain the mob from.

Ultimately changing this will mean we waste the construction of cause_data in the parent because we need to redo it if we don't provide those three arguments to the parent constructor.

Reply:
Currently reverted this change since I think its better to reduce the amount of times cause_data is constructed.